src/rigid_optimization/sdf_gradient_wrt_transformation2d.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * sdf_gradient_wrt_transformation2d.cpp
+ *
+ *  Created on: Mar 05, 2019
+ *      Author: Fei Shan
+ */
+
+//libraries
+#include "Eigen/Eigen"
+
+//local
+#include "../math/tensor_operations.hpp"
+#include "../math/gradients.hpp"
+#include "../math/transformation.hpp"
+
+
+namespace eig = Eigen;
+
+namespace rigid_optimization {
+    void gradient_wrt_twist(const eig::MatrixXf& live_field,
+                            const eig::Vector3f& twist,
+                            const eig::Vector3i& array_offset,
+                            const float& voxel_size,
+                            eig::Matrix<eig::Vector3f, eig::Dynamic, eig::Dynamic>& gradient_field) {
+
+        math::MatrixXv2f gradient_first_term;
+        math::gradient(gradient_first_term, live_field);
+        eig::Matrix3f twist_matrix_homo_inv2d = math::transformation_vector_to_matrix2d(-twist);
+
+        float x_voxel, z_voxel, w_voxel = 1;
+
+        for (int y_field=0; y_field<live_field.rows(); ++y_field) {
+            for (int x_field=0; x_field<live_field.cols(); ++x_field) {
```

Comment:
Same comment as above; it makes more sense to do column-major traversal with Eigen

Reply:
fixed